### PR TITLE
Add notebook for text classification with BERT and text classification tokenization caching

### PR DESCRIPTION
**Pull request by ericharper**
Add notebooks detailing text classification with BERT using NeMo.

[text_classification_with_bert.ipynb](https://github.com/ericharper/NeMo/blob/text-classification-notebook/examples/nlp/text_classification/text_classification_with_bert.ipynb)

[text_classification_with_bert_inference.ipynb](https://github.com/ericharper/NeMo/blob/text-classification-notebook/examples/nlp/text_classification/text_classification_with_bert_inference.ipynb)

We also add a script that converts the text classification tokenization to hdf5: [preproc_text_data.py](https://github.com/ericharper/NeMo/blob/text-classification-notebook/examples/nlp/text_classification/preproc_text_data.py)

We then modify the [text classification dataset](https://github.com/ericharper/NeMo/blob/text-classification-notebook/nemo/collections/nlp/data/datasets/text_classification_dataset.py) to include a preproc version and we modify the [text classification data layer](https://github.com/ericharper/NeMo/blob/text-classification-notebook/nemo/collections/nlp/nm/data_layers/text_classification_datalayer.py) to also have a preproc version. 

The text classification preproc data layer is similar to  [bert pretraining preproc data layer](https://github.com/ericharper/NeMo/blob/0bc3c7a902e2a5a9887d243549388204219a3038/nemo/collections/nlp/nm/data_layers/lm_bert_datalayer.py#L78)

We found it more efficient to cache the tokenization when training. We were able to get better training times and better scaling when using multiple GPUs.

The notebooks also visualize the BERT embeddings before and after fine-tuning by applying T-SNE.

The inference notebooks allow the user to see how well the fine-tuned models are classifying their data.

We also include a script to download the SST-2 dataset for use in the notebook.

We changed the text classification dataset to use the user specified max seq length.

### CODE REVIEW

**Comment by okuchaiev** [comment]
can you please sign your commits first? The DCO's bot output shows you how it can be done https://github.com/NVIDIA/NeMo/pull/382/checks?check_run_id=454165747

**Comment by ericharper** [comment]
Okay, I was able to sign off on the commits.

**Comment by okuchaiev** [comment]
@ericharper there were some changes with how NLP collection is organized (and I think, for now, we are done with changes there).
Could you please rebase your PR on the latest master?

**Comment by ericharper** [comment]
Rebased to master.

**Comment by okuchaiev** [comment]
looks fine to me, @ekmb can you please have a look too?  @ericharper is this still WIP?

**Comment by ericharper** [comment]
finished on my end, but happy to make more changes if needed

File: nemo/collections/nlp/data/datasets/text_classification_dataset.py [hunk 1]
```diff
@@ -52,68 +61,102 @@ class BertTextClassificationDataset(Dataset):
...
     def __len__(self):
-        return len(self.features)
+        if self.use_cache:
+            return len(self.features[0])
+
+        else:
+            return len(self.features)
 
     def __getitem__(self, idx):
+        if self.use_cache:
```
Comment (ekmb) [inline nemo/collections/nlp/data/datasets/text_classification_dataset.py]: @VahidooX what do you think about this set up?

Comment (VahidooX) [reply nemo/collections/nlp/data/datasets/text_classification_dataset.py]: I think it should be ok for now. We can merge it into nemo and improve it later.

**Comment by ekmb** [comment]
@ericharper Thank you for your contribution!

**Pull request merged**